mrchecker-framework-modules/mrchecker-core-module/src/test/java/com/capgemini/mrchecker/test/core/utils/encryption/DataEncryptionGUITest.java

```diff
@@ -0,0 +1,325 @@
+package com.capgemini.mrchecker.test.core.utils.encryption;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.core.StringStartsWith.startsWith;
+
+import java.awt.*;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import org.assertj.swing.core.EmergencyAbortListener;
+import org.assertj.swing.edt.FailOnThreadViolationRepaintManager;
+import org.assertj.swing.edt.GuiActionRunner;
+import org.assertj.swing.fixture.FrameFixture;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.parallel.ResourceLock;
+
+import com.capgemini.mrchecker.test.core.tags.IntegrationTest;
+import com.capgemini.mrchecker.test.core.utils.encryption.controller.DataEncryptionController;
+import com.capgemini.mrchecker.test.core.utils.encryption.controller.IDataEncryptionController;
+import com.capgemini.mrchecker.test.core.utils.encryption.view.DataEncryptionGUI;
+import com.capgemini.mrchecker.test.core.utils.encryption.view.IDataEncryptionView;
+
+@IntegrationTest
+@ResourceLock(value = "SingleThread")
+public class DataEncryptionGUITest {
+	private static final IDataEncryptionController dataEncryptionController = new DataEncryptionController();
+	private FrameFixture dataEncryptionGUIFixture;
+	private EmergencyAbortListener listener;
+	private static final String secret = "123456789";
+	private static final String value = "password";
+	private static final String cipher = "ENC(nC9VLcpRUZcMOQGqruRSs3D+cKlZ6Ohl)";
+	private static final String cipherForSecretDataFile = "ENC(baAZgn+nIcS0eXFLaRiVVGqxjbtt9tP7)";
+	
+	@BeforeAll
+	public static void setupBeforeAll() {
+		FailOnThreadViolationRepaintManager.install();
+	}
+	
+	@BeforeEach
+	public void setupBeforeEveryTest() {
+		IDataEncryptionView dataEncryptionGUI = GuiActionRunner.execute(() -> new DataEncryptionGUI(dataEncryptionController));
+		dataEncryptionController.setView(dataEncryptionGUI);
+		listener = EmergencyAbortListener.registerInToolkit();
+		dataEncryptionGUIFixture = new FrameFixture((Frame) dataEncryptionGUI);
+	}
+	
+	@Test
+	public void shouldEncryptDataWithGivenSecretAndValue() {
+		CryptParams cryptParams = new CryptParams(secret, value);
+		dataEncryptionGUIFixture.textBox("encryptionKeyTextField")
+				.enterText(cryptParams.getSecret());
+		dataEncryptionGUIFixture.textBox("encryptionTextField")
+				.enterText(cryptParams.getText());
+		dataEncryptionGUIFixture.button("encryptButton")
+				.click();
+		String result = dataEncryptionGUIFixture.textBox("encryptionResultTextField")
+				.text();
+		
+		assertThat(result, startsWith("ENC("));
```

Comment:
"ENC(" is coupled with DataEncrytpionController. Use mock instead and define return value.